### PR TITLE
Feature: add form urlencoded & raw request body options

### PR DESCRIPTION
**Pull request by akbarcandra**
Checklist:
- [x] Add form url encoded body option
- [x] Add raw body option (text, JSON, XML, YAML, EDN) using [CodeMirror](https://codemirror.net/)
- [x] Refactor request tabs component to use `sync` modifier for `request` prop

Preview
image
image
image
image

### CODE REVIEW

**Comment by akbarcandra** [comment]
#30

**Comment by davidhsianturi** [comment]
@akbarcandra thanks for sending this in! It's been pretty hectic for me lately so it might be a while before I get to this. But I haven't forgotten about it yet, just so you know :)

**Comment by akbarcandra** [comment]
Okay no problem

**Comment by davidhsianturi** [comment]
@akbarcandra this is nice additions and the API call still works like a charm 👍
I noticed that you are using the same table row in different form options, 
are we expecting this UX?
Nov-17-2019 23-19-55

File: resources/js/components/RequestTabs.vue [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+import BodyRawVue from './request/BodyRaw';
 
 export default {
     components: {
         'dropdown': Dropdown,
-        'files-input': FilesInput,
         'header-fields': HeaderFields,
+        'body-options': BodyOptions,
+        'body-multipart-form': BodyMultipartForm,
+        'body-from-url-encoded': BodyFormUrlEncoded,
```
Comment (davidhsianturi) [inline resources/js/components/RequestTabs.vue]: I guess the word `from` is a typo. let's change this to `body-form-url-encoded` ?

Comment (akbarcandra) [reply resources/js/components/RequestTabs.vue]: Oops, sorry 😅
Fixed

**Comment by akbarcandra** [comment]
> @akbarcandra this is nice additions and the API call still works like a charm 👍
> I noticed that you are using the same table row in different form options,
> are we expecting this UX?
> Nov-17-2019 23-19-55

Hmm, you right, that will cause a problem if using same table row for different form options, especially for **File** value. I will try to fix this by separating it.

**Comment by akbarcandra** [comment]
@davidhsianturi have you tried to test for **Raw** options? Especially for **XML, YAML, EDN**. Is it working to be received by an API? If it is to hard to test, I can remove that feature for this PR and put in another PR.

**Comment by davidhsianturi** [comment]
> Hmm, you right, that will cause a problem if using same table row for different form options, especially for **File** value. I will try to fix this by separating it.

Maybe you can add the identifier like `id` to the raw? the raw table should be unique I guess

**Comment by davidhsianturi** [comment]
> @davidhsianturi have you tried to test for **Raw** options? Especially for **XML, YAML, EDN**. Is it working to be received by an API? If it is to hard to test, I can remove that feature for this PR and put in another PR.

right, I think we just need the JSON options for now.

**Comment by akbarcandra** [comment]
Hi @davidhsianturi , I already push some fixes, please to review, thanks 😊

**Pull request merged**